Index: app/src/main/java/com/example/myapplication/ListActivity.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.BaseRevisionTextPatchEP
<+>package com.example.myapplication;\n\nimport androidx.appcompat.app.ActionBar;\nimport androidx.appcompat.app.AppCompatActivity;\nimport androidx.recyclerview.widget.LinearLayoutManager;\nimport androidx.recyclerview.widget.RecyclerView;\n\nimport android.os.Bundle;\n\npublic class ListActivity extends AppCompatActivity {\n\n\n    @Override\n    protected void onCreate(Bundle savedInstanceState) {\n        super.onCreate(savedInstanceState);\n        setContentView(R.layout.activity_list);\n\n        ActionBar actionBar = getSupportActionBar();\n        actionBar.hide();\n\n\n\n    }\n}
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/app/src/main/java/com/example/myapplication/ListActivity.java b/app/src/main/java/com/example/myapplication/ListActivity.java
--- a/app/src/main/java/com/example/myapplication/ListActivity.java	(revision defca7c2c256a62d4d7cfe08415c8a656f07f7bd)
+++ b/app/src/main/java/com/example/myapplication/ListActivity.java	(date 1652890023289)
@@ -6,6 +6,9 @@
 import androidx.recyclerview.widget.RecyclerView;
 
 import android.os.Bundle;
+import android.provider.ContactsContract;
+
+import java.util.ArrayList;
 
 public class ListActivity extends AppCompatActivity {
 
@@ -18,7 +21,17 @@
         ActionBar actionBar = getSupportActionBar();
         actionBar.hide();
 
+        RecyclerViewAdapter adapter;
+            RecyclerView recyclerView;
 
+            //데이터 모델 리스트
+        ArrayList<DataModel> dataModels = new ArrayList();
+        dataModels.add(new DataModel("서일분식","서일점",R.drawable.ddug));
 
     }
+
+    private class DataModel {
+        public DataModel(String 서일분식, String 서일점, int ddug) {
+        }
+    }
 }
\ No newline at end of file
Index: app/src/main/java/com/example/myapplication/RecyclerViewAdapter.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/app/src/main/java/com/example/myapplication/RecyclerViewAdapter.java b/app/src/main/java/com/example/myapplication/RecyclerViewAdapter.java
new file mode 100644
--- /dev/null	(date 1652889568528)
+++ b/app/src/main/java/com/example/myapplication/RecyclerViewAdapter.java	(date 1652889568528)
@@ -0,0 +1,69 @@
+package com.example.myapplication;
+
+import android.content.Context;
+import android.provider.ContactsContract;
+import android.view.ViewGroup;
+
+import androidx.annotation.NonNull;
+import androidx.recyclerview.widget.RecyclerView;
+
+import java.sql.DatabaseMetaData;
+import java.util.ArrayList;
+
+public class RecyclerViewAdapter extends RecyclerView.Adapter {
+    /*어댑터의 동작원리 순서
+     1.(getItemCount) 데이터 개수를 세어 어댑터가 만들어야 할 총 아이템 개수를 얻는다
+     2.(getItemViewType) itemView의 viewType을 얻는다 (생략가능)
+     3.(onCreateViewHolder) viewtype에 맞는 뷰 홀더를 생성하여 onBindViewHolder에 전달
+     4.(onBindViewHolder) 뷰홀더와 position을 받아 position에 맞는 데이터를 뷰홀더의 뷰 들에 바인딩한다.*/
+    String Tag = "RecyclerViewAdapter";
+
+    //리사이클러뷰에 넣을 데이터 리스트
+    ArrayList<DataModel> dataModels;
+    Context context;
+
+    //생성자를 통하여 데이터리스트 context를 받음
+
+    public RecyclerViewAdapter(Context context, ArrayList<DataModel> dataModels){
+        this.dataModels = dataModels;
+        this.context = context;
+    }
+
+    @NonNull
+    @Override
+    public RecyclerView.ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
+        return null;
+    }
+
+    @Override
+    public void onBindViewHolder(@NonNull RecyclerView.ViewHolder holder, int position) {
+
+    }
+
+    @Override
+    public int getItemCount() {
+        return dataModels.size();
+    }
+
+    private class DataModel {
+        String title;
+        int image_path;
+
+        public String getTitle(){
+            return title;
+        }
+        public void setTitle(String title){
+            this.title=title;
+        }
+        public int getImage_path(){
+            return image_path;
+        }
+        public void setImage_path(){
+            this.image_path = image_path;
+        }
+        public DataModel(String title, int image_path){
+            this.title = title;
+            this.image_path = image_path;
+        }
+    }
+}
